<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
<title>org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException():v.45a2754f89c8ea8a5d2b4deeb88b1440_node8</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<style type="text/css">
body { background-color:#fff; color:#000; font-family:Helvetica, sans-serif; font-size:10pt }
h1 { font-size:14pt }
.code { border-collapse:collapse; width:100%; font-family: "Andale Mono", monospace; font-size:10pt; line-height: 1.2em }
.color_black { color: black }
.color_blue { color: blue }
.color_green { color: green }
.color_orange { color: orange }
.color_red { color: red }
.line { border-left: 3px solid #ccc; padding-left: 1ex; white-space: pre; }
.msg { background-color:#fff8b4; color:#000; float:left; font-family:Helvetica, sans-serif; font-size: smaller; font-weight: bold; margin-bottom:10px; margin-top:10px; max-width:60em; padding:0.5em 1ex 0.5em 1ex; -webkit-border-radius:5px; -webkit-box-shadow:1px 1px 7px #000; white-space: pre-wrap; word-wrap: break-word; }
.num { background-color:#eee; color:#444; font-size: smaller; padding-right:2ex; text-align:right; width:2.5em; }
.dangling { background-color:white; color: gray; }
.visited { background-color:LemonChiffon; color: darkmagenta; }
.tooltip { display: none; background-color:#FFF0F0; border: 2px solid #F00; font-weight: normal; left:10em; padding: 2px; position: absolute; top: -1em; -webkit-border-radius:5px; -webkit-box-shadow:1px 1px 7px #000; z-index: 1}
.with_tooltip { position: relative; }
.with_tooltip:hover .tooltip, .visited:hover .tooltip { display: block; }
</style>
</head>
<body>
<center><h1>Cfg Node <a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">8</a></h1></center>PROC: <a href="./../org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440.html">void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException()</a> LINE: <a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">43</a>
<br>PREDS:
<a href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node11.html"><span class='visited with_tooltip'>N_11<span class='tooltip'>node11 preds:13 succs:8 exn:3 </span></span></a><br>SUCCS:
<a href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node9.html"><span class='visited with_tooltip'>N_9<span class='tooltip'>node9 preds:8 succs:10 exn:3 </span></span></a><br>EXN:
<a href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node3.html"><span class='visited with_tooltip'>N_3<span class='tooltip'>node3 preds:12 succs:2 exn:2 </span></span></a><br>
<LISTING class='color_green'>n$7=*&amp;CatchVar25:java.lang.Object* [line 43];
*&amp;e:java.lang.Object*=n$7 [line 43];
</LISTING>
<hr width="100%">
<a name="session8" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node8.html#session8">node8#session8</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">43</a>) [compute pre] liveness 8
<LISTING class='color_black'>JOIN

LEFT:   normal:{ exception, e }, exn:{ }
RIGHT:  normal:{ }, exn:{ }
RESULT: normal:{ exception, e }, exn:{ }

</LISTING>
<hr width="100%">
<a name="session9" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node8.html#session9">node8#session9</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">43</a>) [exec] liveness 8
<LISTING class='color_black'>PRE STATE:
normal:{ exception, e }, exn:{ }

INSTR=  *&amp;e:java.lang.Object*=n$7 [line 43]; 

STATE:
normal:{ n$7, exception }, exn:{ }

INSTR=  n$7=*&amp;CatchVar25:java.lang.Object* [line 43]; 

STATE:
normal:{ exception, CatchVar25 }, exn:{ }

</LISTING>
<hr width="100%">
<a name="session39" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node8.html#session39">node8#session39</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">43</a>) [compute pre] nullify
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session40" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node8.html#session40">node8#session40</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">43</a>) [exec] nullify
<LISTING class='color_black'>PRE STATE:
(normal:{ exception, CatchVar25 }, exn:{ },normal:{ n$12, n$13 }, exn:{ })

INSTR=  n$7=*&amp;CatchVar25:java.lang.Object* [line 43]; 

STATE:
(normal:{ n$7, exception, CatchVar25 }, exn:{ },
normal:{ n$12, n$13 }, exn:{ })

INSTR=  *&amp;e:java.lang.Object*=n$7 [line 43]; 

STATE:
(normal:{ exception, e }, exn:{ },normal:{ n$7, CatchVar25 }, exn:{ })

</LISTING>
<hr width="100%">
<a name="session87" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node8.html#session87">node8#session87</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">43</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session88" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node8.html#session88">node8#session88</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">43</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session89" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node8.html#session89">node8#session89</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">43</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session90" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node8.html#session90">node8#session90</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">43</a>) [exec] devirtualizer
<LISTING class='color_black'></LISTING>
<hr width="100%">
<a name="session34" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node8.html#session34">node8#session34</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">43</a>) biabduction
<LISTING class='color_black'>**** [FP:Pending] void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException() Node: 8, Procedure: void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException(), Todo: 2 ****
  PROP 1 of 2:
  <span class='color_blue'><span class='color_red'>(-1  &lt;  @f$1)</span>; <span class='color_red'>CatchVar25 != 0</span>; <span class='color_red'>@f$1 != 1</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; Object List.get(int) &gt; :92(CatchVar25)</span>; <span class='color_red'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
  <span class='color_red'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_red'>CatchVar25 = _t$1<span class='color_orange'>update:43</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_red'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_red'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>old_this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>e = _t$2<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
  <span class='color_red'>CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
  [footprint
     <span class='color_red'>(-1  &lt;  @f$1)</span>; <span class='color_red'>@f$1 != 1</span> *
     <span class='color_red'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_red'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
     <span class='color_red'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
  PROP 2 of 2:
  <span class='color_blue'><span class='color_red'>(@f$2  &lt;=  -1)</span>; <span class='color_red'>CatchVar25 != 0</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; Object List.get(int) &gt; :92(CatchVar25)</span>; <span class='color_red'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
  <span class='color_red'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_red'>CatchVar25 = _t$1<span class='color_orange'>update:43</span>:void</span>; <span class='color_red'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_red'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_red'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>e = _t$2<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
  <span class='color_red'>CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>@f$0|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
  [footprint
     <span class='color_red'>(@f$2  &lt;=  -1)</span> *
     <span class='color_red'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_red'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
     <span class='color_red'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
  .... Instructions: ....
  <span class='color_green'>n$7=*&amp;CatchVar25:java.lang.Object* [line 43];
  *&amp;e:java.lang.Object*=n$7 [line 43];
  NULLIFY(&amp;CatchVar25); [line 43];
  EXIT_SCOPE(n$7,CatchVar25); [line 43];
  </span>
  
  Processing prop 1/2
    Instruction <span class='color_green'>n$7=*&amp;CatchVar25:java.lang.Object* [line 43]</span>
    in vpath_find exp:n$181
    vpath_find: found exception :java.lang.Throwable*
    .... Rearrangement Start ....
    Exp: &amp;CatchVar25
    Prop:
    <span class='color_blue'>(-1  &lt;  @f$1); CatchVar25 != 0; @f$1 != 1; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; Object List.get(int) &gt; :92(CatchVar25); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$181<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$180<span class='color_orange'>update:43</span>:void; return = null<span class='color_orange'>update:43</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$182<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub ); @f$0|->{mIsEmpty:@f$1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       (-1  &lt;  @f$1); @f$1 != 1 *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:@f$1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    
    in prop_iter_extend_ptsto lexp: &amp;CatchVar25
    prop before:
    <span class='color_blue'>(-1  &lt;  @f$1); CatchVar25 != 0; @f$1 != 1; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; Object List.get(int) &gt; :92(CatchVar25); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$181<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$180<span class='color_orange'>update:43</span>:void; return = null<span class='color_orange'>update:43</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$182<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub ); @f$0|->{mIsEmpty:@f$1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       (-1  &lt;  @f$1); @f$1 != 1 *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:@f$1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    
    
    prop list after:
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_blue'>CatchVar25 != 0</span>; <span class='color_blue'>@f$1 != 1</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(CatchVar25)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = n$181<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = n$180<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = n$182<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_blue'>@f$1 != 1</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    
    
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_red'>CatchVar25 != 0</span>; <span class='color_blue'>@f$1 != 1</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; Object List.get(int) &gt; :92(CatchVar25)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>n$7<span class='color_orange'>update:43</span></span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$1<span class='color_orange'>initial</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_blue'>@f$1 != 1</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>*&amp;e:java.lang.Object*=n$7 [line 43]</span>
    in vpath_find exp:n$183
    vpath_find: found exception :java.lang.Throwable*
    .... Rearrangement Start ....
    Exp: &amp;e
    Prop:
    <span class='color_blue'>(-1  &lt;  @f$1); CatchVar25 != 0; @f$1 != 1; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; Object List.get(int) &gt; :92(CatchVar25); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$183<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$7<span class='color_orange'>update:43</span>:void; return = null<span class='color_orange'>update:43</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$184<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub ); @f$0|->{mIsEmpty:@f$1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       (-1  &lt;  @f$1); @f$1 != 1 *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:@f$1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    
    in prop_iter_extend_ptsto lexp: &amp;e
    prop before:
    <span class='color_blue'>(-1  &lt;  @f$1); CatchVar25 != 0; @f$1 != 1; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; Object List.get(int) &gt; :92(CatchVar25); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$183<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$7<span class='color_orange'>update:43</span>:void; return = null<span class='color_orange'>update:43</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$184<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub ); @f$0|->{mIsEmpty:@f$1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       (-1  &lt;  @f$1); @f$1 != 1 *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:@f$1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    
    
    prop list after:
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_blue'>CatchVar25 != 0</span>; <span class='color_blue'>@f$1 != 1</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(CatchVar25)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = n$183<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = n$7<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = n$184<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_blue'>@f$1 != 1</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    
    
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_blue'>e != 0</span>; <span class='color_blue'>@f$1 != 1</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = n$7<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>n$7<span class='color_orange'>update:43</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_blue'>@f$1 != 1</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>NULLIFY(&amp;CatchVar25); [line 43]</span>
    in vpath_find exp:_t$0
    vpath_find: found exception :java.lang.Throwable*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_blue'>e != 0</span>; <span class='color_blue'>@f$1 != 1</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = n$7<span class='color_orange'>update:43</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_blue'>@f$1 != 1</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$7,CatchVar25); [line 43]</span>
    in vpath_find exp:_t$0
    vpath_find: found exception :java.lang.Throwable*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_red'>e != 0</span>; <span class='color_blue'>@f$1 != 1</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$1<span class='color_orange'>update:43</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_blue'>@f$1 != 1</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_blue'>e != 0</span>; <span class='color_blue'>@f$1 != 1</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$1<span class='color_orange'>update:43</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(-1  &lt;  @f$1)</span>; <span class='color_blue'>@f$1 != 1</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$2<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$1<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    
    
  Processing prop 2/2
    Instruction <span class='color_green'>n$7=*&amp;CatchVar25:java.lang.Object* [line 43]</span>
    in vpath_find exp:n$186
    vpath_find: found exception :java.lang.Throwable*
    .... Rearrangement Start ....
    Exp: &amp;CatchVar25
    Prop:
    <span class='color_blue'>(@f$2  &lt;=  -1); CatchVar25 != 0; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; Object List.get(int) &gt; :92(CatchVar25); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$186<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$185<span class='color_orange'>update:43</span>:void; return = null<span class='color_orange'>update:43</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$187<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub ); @f$0|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       (@f$2  &lt;=  -1) *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    
    in prop_iter_extend_ptsto lexp: &amp;CatchVar25
    prop before:
    <span class='color_blue'>(@f$2  &lt;=  -1); CatchVar25 != 0; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; Object List.get(int) &gt; :92(CatchVar25); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$186<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$185<span class='color_orange'>update:43</span>:void; return = null<span class='color_orange'>update:43</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$187<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub ); @f$0|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       (@f$2  &lt;=  -1) *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    
    
    prop list after:
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(@f$2  &lt;=  -1)</span>; <span class='color_blue'>CatchVar25 != 0</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(CatchVar25)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = n$186<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = n$185<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = n$187<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>@f$0|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(@f$2  &lt;=  -1)</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    
    
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(@f$2  &lt;=  -1)</span>; <span class='color_red'>CatchVar25 != 0</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; Object List.get(int) &gt; :92(CatchVar25)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>n$7<span class='color_orange'>update:43</span></span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$1<span class='color_orange'>initial</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>@f$0|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(@f$2  &lt;=  -1)</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>*&amp;e:java.lang.Object*=n$7 [line 43]</span>
    in vpath_find exp:n$188
    vpath_find: found exception :java.lang.Throwable*
    .... Rearrangement Start ....
    Exp: &amp;e
    Prop:
    <span class='color_blue'>(@f$2  &lt;=  -1); CatchVar25 != 0; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; Object List.get(int) &gt; :92(CatchVar25); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$188<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$7<span class='color_orange'>update:43</span>:void; return = null<span class='color_orange'>update:43</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$189<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub ); @f$0|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       (@f$2  &lt;=  -1) *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    
    in prop_iter_extend_ptsto lexp: &amp;e
    prop before:
    <span class='color_blue'>(@f$2  &lt;=  -1); CatchVar25 != 0; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; Object List.get(int) &gt; :92(CatchVar25); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0) ;
    exception = n$188<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$7<span class='color_orange'>update:43</span>:void; return = null<span class='color_orange'>update:43</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$189<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub ); @f$0|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )
    [footprint
       (@f$2  &lt;=  -1) *
       $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub ); this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ) ;
       @f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )  ]</span>
    
    
    prop list after:
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(@f$2  &lt;=  -1)</span>; <span class='color_blue'>CatchVar25 != 0</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(CatchVar25)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = n$188<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = n$7<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = n$189<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>@f$0|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(@f$2  &lt;=  -1)</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    
    
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(@f$2  &lt;=  -1)</span>; <span class='color_blue'>e != 0</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = n$7<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>n$7<span class='color_orange'>update:43</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>@f$0|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(@f$2  &lt;=  -1)</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>NULLIFY(&amp;CatchVar25); [line 43]</span>
    in vpath_find exp:_t$0
    vpath_find: found exception :java.lang.Throwable*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(@f$2  &lt;=  -1)</span>; <span class='color_blue'>e != 0</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = n$7<span class='color_orange'>update:43</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>@f$0|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(@f$2  &lt;=  -1)</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$7,CatchVar25); [line 43]</span>
    in vpath_find exp:_t$0
    vpath_find: found exception :java.lang.Throwable*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(@f$2  &lt;=  -1)</span>; <span class='color_red'>e != 0</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$1<span class='color_orange'>update:43</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>@f$0|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(@f$2  &lt;=  -1)</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(@f$2  &lt;=  -1)</span>; <span class='color_blue'>e != 0</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39(@f$0)</span> ;
    <span class='color_blue'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$1<span class='color_orange'>update:43</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>@f$0|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>
    [footprint
       <span class='color_blue'>(@f$2  &lt;=  -1)</span> *
       <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = @f$0<span class='color_orange'>none</span>:java.util.List*( sub )</span>; <span class='color_blue'>this = @f$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span> ;
       <span class='color_blue'>@f$0|->{mIsEmpty:@f$2<span class='color_orange'>rearrange:(z)27</span>}<span class='color_orange'>rearrange:(z)88</span>:java.util.List( sub )</span>  ]</span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session51" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node8.html#session51">node8#session51</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">43</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException() Node: 8, Procedure: void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>(-1  &lt;  val$3)</span>; <span class='color_red'>CatchVar25 != 0</span>; <span class='color_red'>val$3 != 1</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; Object List.get(int) &gt; :92(CatchVar25)</span>; <span class='color_red'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
  <span class='color_red'>exception = _t$2<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_red'>CatchVar25 = _t$1<span class='color_orange'>update:43</span>:void</span>; <span class='color_red'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_red'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_red'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>e = _t$0<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
  <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_red'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:val$3<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
  .... Instructions: ....
  <span class='color_green'>n$7=*&amp;CatchVar25:java.lang.Object* [line 43];
  *&amp;e:java.lang.Object*=n$7 [line 43];
  NULLIFY(&amp;CatchVar25); [line 43];
  EXIT_SCOPE(n$7,CatchVar25); [line 43];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$7=*&amp;CatchVar25:java.lang.Object* [line 43]</span>
    in vpath_find exp:n$109
    vpath_find: found exception :java.lang.Throwable*
    .... Rearrangement Start ....
    Exp: &amp;CatchVar25
    Prop:
    <span class='color_blue'>(-1  &lt;  val$3); val$3 != 1; CatchVar25 != 0; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; Object List.get(int) &gt; :92(CatchVar25); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
    exception = n$109<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$110<span class='color_orange'>update:43</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = null<span class='color_orange'>update:43</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$111<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:val$3<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(-1  &lt;  val$3)</span>; <span class='color_blue'>val$3 != 1</span>; <span class='color_red'>CatchVar25 != 0</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; Object List.get(int) &gt; :92(CatchVar25)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>n$7<span class='color_orange'>update:43</span></span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$0<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:val$3<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>; <span class='color_red'>CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span></span>
    
    Instruction <span class='color_green'>*&amp;e:java.lang.Object*=n$7 [line 43]</span>
    in vpath_find exp:n$112
    vpath_find: found exception :java.lang.Throwable*
    .... Rearrangement Start ....
    Exp: &amp;e
    Prop:
    <span class='color_blue'>(-1  &lt;  val$3); val$3 != 1; CatchVar25 != 0; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; Object List.get(int) &gt; :92(CatchVar25); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
    exception = n$112<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$7<span class='color_orange'>update:43</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = null<span class='color_orange'>update:43</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$113<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:val$3<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(-1  &lt;  val$3)</span>; <span class='color_blue'>val$3 != 1</span>; <span class='color_blue'>e != 0</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$0<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = n$7<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>n$7<span class='color_orange'>update:43</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:val$3<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>; <span class='color_blue'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span></span>
    
    Instruction <span class='color_green'>NULLIFY(&amp;CatchVar25); [line 43]</span>
    in vpath_find exp:_t$0
    vpath_find: found exception :java.lang.Throwable*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(-1  &lt;  val$3)</span>; <span class='color_blue'>val$3 != 1</span>; <span class='color_blue'>e != 0</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = n$7<span class='color_orange'>update:43</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:val$3<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>; <span class='color_blue'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$7,CatchVar25); [line 43]</span>
    in vpath_find exp:_t$0
    vpath_find: found exception :java.lang.Throwable*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(-1  &lt;  val$3)</span>; <span class='color_blue'>e != 0</span>; <span class='color_blue'>val$3 != 1</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$0<span class='color_orange'>update:43</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:val$3<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(-1  &lt;  val$3)</span>; <span class='color_red'>e != 0</span>; <span class='color_blue'>val$3 != 1</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$0<span class='color_orange'>update:43</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:val$3<span class='color_orange'>formal(z)</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
    
    
    
  </LISTING>
<hr width="100%">
<a name="session64" href="./../nodes/org.junit.tests.assertion.MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException%28%29%3Av.45a2754f89c8ea8a5d2b4deeb88b1440_node8.html#session64">node8#session64</a>(<a href="./../../MultipleFailureExceptionTest.java.5c3e6d8739a7da2006aef2128f5f1965.html#LINE43">43</a>) biabduction
<LISTING class='color_black'>**** [RE:Pending] void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException() Node: 8, Procedure: void MultipleFailureExceptionTest.assertEmptyRethrowsSingleRuntimeException(), Todo: 1 ****
  PROP 1 of 1:
  <span class='color_blue'><span class='color_red'>(val$3  &lt;=  -1)</span>; <span class='color_red'>CatchVar25 != 0</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; Object List.get(int) &gt; :92(CatchVar25)</span>; <span class='color_red'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
  <span class='color_red'>exception = _t$2<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_red'>CatchVar25 = _t$1<span class='color_orange'>update:43</span>:void</span>; <span class='color_red'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_red'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_red'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_red'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_red'>e = _t$0<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
  <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_red'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
  .... Instructions: ....
  <span class='color_green'>n$7=*&amp;CatchVar25:java.lang.Object* [line 43];
  *&amp;e:java.lang.Object*=n$7 [line 43];
  NULLIFY(&amp;CatchVar25); [line 43];
  EXIT_SCOPE(n$7,CatchVar25); [line 43];
  </span>
  
  Processing prop 1/1
    Instruction <span class='color_green'>n$7=*&amp;CatchVar25:java.lang.Object* [line 43]</span>
    in vpath_find exp:n$224
    vpath_find: found exception :java.lang.Throwable*
    .... Rearrangement Start ....
    Exp: &amp;CatchVar25
    Prop:
    <span class='color_blue'>(val$3  &lt;=  -1); CatchVar25 != 0; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; Object List.get(int) &gt; :92(CatchVar25); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
    exception = n$224<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$225<span class='color_orange'>update:43</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = null<span class='color_orange'>update:43</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$226<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(val$3  &lt;=  -1)</span>; <span class='color_red'>CatchVar25 != 0</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; Object List.get(int) &gt; :92(CatchVar25)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>n$7<span class='color_orange'>update:43</span></span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$0<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>; <span class='color_red'>CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span></span>
    
    Instruction <span class='color_green'>*&amp;e:java.lang.Object*=n$7 [line 43]</span>
    in vpath_find exp:n$227
    vpath_find: found exception :java.lang.Throwable*
    .... Rearrangement Start ....
    Exp: &amp;e
    Prop:
    <span class='color_blue'>(val$3  &lt;=  -1); CatchVar25 != 0; MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception); UND &lt; Object List.get(int) &gt; :92(CatchVar25); UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar) ;
    exception = n$227<span class='color_orange'>update:38</span>:java.lang.Throwable*; CatchVar25 = n$7<span class='color_orange'>update:43</span>:void; $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub ); return = null<span class='color_orange'>update:43</span>:void; errors = 0<span class='color_orange'>nullify</span>:java.util.List*; this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub ); e = n$228<span class='color_orange'>initial</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException* ;
    $RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub ); exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException; CatchVar25|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>
    
    Prover.check_inconsistency_base: inconsistency reason [None]
    Prover.check_inconsistency_base: inconsistency reason [None]
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(val$3  &lt;=  -1)</span>; <span class='color_blue'>e != 0</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$0<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = n$7<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>n$7<span class='color_orange'>update:43</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>; <span class='color_blue'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span></span>
    
    Instruction <span class='color_green'>NULLIFY(&amp;CatchVar25); [line 43]</span>
    in vpath_find exp:_t$0
    vpath_find: found exception :java.lang.Throwable*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(val$3  &lt;=  -1)</span>; <span class='color_blue'>e != 0</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = _t$0<span class='color_orange'>update:38</span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = n$7<span class='color_orange'>update:43</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span>; <span class='color_blue'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span></span>
    
    Instruction <span class='color_green'>EXIT_SCOPE(n$7,CatchVar25); [line 43]</span>
    in vpath_find exp:_t$0
    vpath_find: found exception :java.lang.Throwable*
    Instruction Returns
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(val$3  &lt;=  -1)</span>; <span class='color_blue'>e != 0</span>; <span class='color_red'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_blue'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = 0<span class='color_orange'>nullify</span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = <span class='color_red'>_t$0<span class='color_orange'>update:43</span></span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_red'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_blue'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
    
    .... After Symbolic Execution ....
    PROP 1 of 1:
    <span class='color_blue'><span class='color_blue'>(val$3  &lt;=  -1)</span>; <span class='color_red'>e != 0</span>; <span class='color_blue'>MEMne &lt; new:38 &gt; <span class='color_orange'>{vpath: }</span>(exception)</span>; <span class='color_red'>UND &lt; Object List.get(int) &gt; :92(e)</span>; <span class='color_blue'>UND &lt; List Collections.singletonList(MultipleFailureExceptionTest$ExpectedException) &gt; :39($RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar)</span> ;
    <span class='color_blue'>exception = <span class='color_red'>_t$1<span class='color_orange'>update:38</span></span>:java.lang.Throwable*</span>; <span class='color_blue'>CatchVar25 = <span class='color_red'>0<span class='color_orange'>nullify</span></span>:void</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar = val$2<span class='color_orange'>formal</span>:java.util.List*( sub )</span>; <span class='color_blue'>return = null<span class='color_orange'>update:43</span>:void</span>; <span class='color_blue'>errors = 0<span class='color_orange'>nullify</span>:java.util.List*</span>; <span class='color_blue'>this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>old_this = val$1<span class='color_orange'>formal</span>:org.junit.tests.assertion.MultipleFailureExceptionTest*( sub )</span>; <span class='color_blue'>e = _t$0<span class='color_orange'>update:43</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException*</span> ;
    <span class='color_blue'>exception|->{}<span class='color_orange'>alloc</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException</span>; <span class='color_red'>e|->{}<span class='color_orange'>none</span>:org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException( sub )</span>; <span class='color_blue'>$RET_java.util.Collections.singletonList(org.junit.tests.assertion.MultipleFailureExceptionTest$ExpectedException):java.util.List|abducedRetvar|->{mIsEmpty:0<span class='color_orange'>update:(z)41</span>}<span class='color_orange'>formal(z)</span>:java.util.List( sub )</span></span>
    
    
    
  </LISTING>